ntbs-service/Models/Validations/ValidationAttributes.cs

```diff
@@ -44,4 +45,47 @@ private bool isTruthy(object value)
             return value != null && (bool)value;
         }
     }
+
+    public class ValidPartialDateCanConvertToDatetimeAttribute : ValidationAttribute
+    {
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            PartialDate valueCopy = (PartialDate) value;
+            if(valueCopy==null || valueCopy.IsEmpty()) {
+                return null;
+            }
+            if (!string.IsNullOrEmpty(valueCopy.Day)) {
+                if(string.IsNullOrEmpty(valueCopy.Month) || string.IsNullOrEmpty(valueCopy.Year)) {
+                    return new ValidationResult("Year and month must be provided if a day has been provided");
+                }
+            }
+            if(!string.IsNullOrEmpty(valueCopy.Month)) {
+                if(string.IsNullOrEmpty(valueCopy.Year)) {
+                    return new ValidationResult("A year must be provided");

```

Comment:
Only one Errormessage can be supplied to this method - these could be static members of this class or left inline here - not sure what is best

Reply:
Can they be static members of the `ValidationMessages` class like the others?